projects/laji/src/app/shared-modules/select/select/select.component.html

```diff
@@ -4,26 +4,36 @@
     <ng-container *ngIf="!useFilter; else multiWithFilter">
       <ng-container *ngIf="!open">
         <span class="item" *ngFor="let option of selectedOptions; trackBy: track">
-          <label class="lj-item selected" (click)="labelClick($event)">
-            <laji-checkbox [value]="true" (valueChange)="remove(option.id)"></laji-checkbox> {{ option.value}}
+          <label class="lj-item selected" [ngClass]="{'subCategoriesOptions': checkboxType !== 2, '': checkboxType === 2}" (click)="labelClick($event)">
+            <ng-container *ngIf="checkboxType === 0 else checkboxSubCategoriesGlobal">
+              <laji-checkbox [value]="true" (valueChange)="remove(option.id, $event)"></laji-checkbox> {{ option.value}}
+            </ng-container>
+            <ng-template #checkboxSubCategoriesGlobal>
+              <laji-checkbox [checkboxType]="2" [value]="selectedOptions | includesMultiState: option : 'id'" (valueChange)="remove(option.id, $event)"></laji-checkbox>{{ option.id | warehouse | label }}
+            </ng-template>
           </label>
           <ng-container [ngTemplateOutlet]="value" [ngTemplateOutletContext]="{option: option}"></ng-container>
         </span>
       </ng-container>
-      <div *ngIf="open" class="lj-item-container">
+      <div *ngIf="open" class="lj-item-container" [ngClass]="{'subCategoriesContainer': checkboxType !== 2, '': checkboxType === 2}">
         <laji-spinner [spinning]="loading"></laji-spinner>
         <div [class.lj-active]="i === selectedIdx" *ngFor="let option of options; let i = index; trackBy: track" style="position: relative;">
-          <label class="lj-item" (click)="labelClick($event)">
-            <laji-checkbox [value]="selectedOptions | includes:option" (valueChange)="toggleValue(option.id)"></laji-checkbox> {{ option.value}}
+          <label class="lj-item" [ngClass]="{'subCategoriesOptions': checkboxType !== 2, '': checkboxType === 2}" (click)="labelClick($event)">
```

Comment:
You can remove the checkboxType === 2 all together since adding empty class to class list is not needed
I would also remove all mentions of `subCategoriesOptions` , `subCategoriesContainer` and `subMenu` from this file and use `@Input`s to get the needed classes, that would make it more usable in other places.

Reply:
now subcategorydata is selectOptions[] and global is selectedOptions[]. Later i filter that with a different pipe